bindings/rust/s2n-tls/src/config.rs

```diff
@@ -291,6 +298,57 @@ impl Builder {
         Ok(self)
     }
 
+    /// Corresponds to [s2n_config_add_cert_chain_and_key_to_store].
+    pub fn load_chain(&mut self, chain: CertificateChain<'static>) -> Result<&mut Self, Error> {
+        // Out of an abudance of caution, we hold a reference to the CertificateChain
+        // regardless of whether add_to_store fails or succeeds. We have limited
+        // visibility into the failure modes, so this behavior ensures that _if_
+        // the C library held the reference despite the failure, it would continue
+        // to be valid memory.
+        self.context_mut()
+            .application_owned_certs
+            .push(chain.clone());
+
+        unsafe {
+            s2n_config_add_cert_chain_and_key_to_store(
+                self.as_mut_ptr(),
+                // SAFETY: audit of add_to_store shows that the certificate chain
+                // is not mutated. https://github.com/aws/s2n-tls/issues/4140
+                chain.as_ptr() as *mut _,
+            )
+            .into_result()
+        }?;
+
+        Ok(self)
+    }
+
+    /// Corresponds to [s2n_config_set_cert_chain_and_key_defaults].
+    pub fn set_default_chains(
+        &mut self,
+        chains: Vec<CertificateChain<'static>>,
+    ) -> Result<&mut Self, Error> {
+        self.context_mut()
+            .application_owned_certs
+            .extend(chains.clone());
+
+        let raw_certs: Vec<*mut s2n_cert_chain_and_key> = chains
```

Comment:
Bummer that we have to allocate here... Is there not a way to append defaults?

Reply:
Unfortunately there isn't a way to append defaults. Each call to `set_defaults` removes the previous defaults, and there is no separate "append" API.

But we can avoid the allocation as long as we are okay with duplicating a constant/some safety checks. This lets us just use an array on the stack instead of a vector. I'll go ahead and implement that in the next revision.